walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,120 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+  	/** General algorithm (first attempt, could be better optimized, generally runs in linear
+     * time except for calls to Collections.disjoint() in the getAllRelevantEvents() method): 
+     * 1. Get a list of all of the events which actually concern the attendees of the request.
+     * 2. Sort the list by start time.
+     * 3. Loop through, and find all available times "between" these events; requires some amount
+     * of shifting around bounds to deal with edge cases regarding nested events.
+     * 4. Return a list of these times, only adding a time if it is longer than the proposed
+     * duration of the MeetingRequest. 
+     */
+     
+    // Run the algorithm the first time also including optional attendees
+  	Collection<Event> relevantEvents = getAllRelevantEvents(events, request, true);
```

Comment:
fix the indentation. Also what is true? If you don't want to create a temp variable for include optional change this to 
getAllRelevantEvents(events, request, /*includeOptional=*/ true);
so the reader can know what true indicates

Reply:
Can you do that in Java?

Reply:
I'll just make a temporary variable.